### DISCUSSION TITLE
PESTPP-IES: ies_bad_phi() option and prior data conflict

### DISCUSSION THREAD

**Discussion by phaehnel** [usgs/pestpp#298] [Q&A] [answered]
This is a more general question as I could not find anything on it in the manual.

In my PESTPP-IES setup for a MODFLOW-based groundwater flow model, I defined `ies_bad_phi(1E30)` to reject realizations that would result in a dry cell at the location of an observation well. The model contains pilot points for hk and vka, initialized with a user-defined random sample, so quite a few runs are rejected due to `ies_bad_phi(1E30)`. When I compare the information from the record file with the .0.obs.csv and .0.par.csv, the realizations rejected due to `ies_bad_phi(1E30)` are part of these files, while the runs that failed due to numerical nonconvergence are not listed there.

I compared the averages and standard deviations for observations from the .0.obs.csv file with the ones reported in the pdc.csv file and they match. So it seems the realizations rejected due to `ies_bad_phi(1E30)` are considered in the calculation of prior data conflict.

So intuitively, I thought they should not be considered for the pdc calculations, just as the realizations where MODFLOW failed numerically. I tried to implement something similar to the workflow reported in Fienen et al. (2022). So I run a prior ensemble and following this update the parameter bounds and values based on the outcomes of this prior ensemble. After some adjustments the prior ensemble did not produce any data conflicts anymore but when I start the history matching with the updated parameter bounds that are based on the prior ensemble excluding the rejected realizations, I get data conflicts again.

Therefore, I am curious if this behavior of considering realizations rejected due to `ies_bad_phi()` in the pdc calculation is intentional and what would be the reasoning for it?

**PEST++ version 5.2.3**

**References**
Fienen, Michael N., Nicholas T. Corson-Dosch, Jeremy T. White, Andrew T. Leaf, und Randall J. Hunt. „Risk-Based Wellhead Protection Decision Support: A Repeatable Workflow Approach“. Groundwater 60, Nr. 1 (2022): 71–86. https://doi.org/10.1111/gwat.13129.

**Comment by jtwhite79** [comment, marked as answer]
> When I compare the information from the record file with the .0.obs.csv and .0.par.csv, the realizations rejected due to ies_bad_phi(1E30) are part of these files, while the runs that failed due to numerical nonconvergence are not listed there.

This is because we want to record the results of all model successful runs even if they are going to be discarded (still might be useful for something) but for failed run, there is nothing to record...

As far as when in the pestpp-ies workflow "bad" realizations are dropped, it happens after conflicts are dealt with - you should see this in the rec file: the PDC part happens before the initial dropping of bad realizations.  I went with this order because I was thinking that if someone is dropping conflicts this could substantially alter the result phi values, which might result in some realizations being kept instead of dropped.  But I can see the other of this also - having extreme simulated values could result in covering an observation that would otherwise be in conflict.   I guess we could apply the absolute bad_phi filter first since its usually a safe guard against implausible conditions, then do the PDC checking, then apply the adaptive bad-phi filter.   Would this fix your problem?

**Reply by phaehnel** [reply]
Thanks for the clarification! Do you mean `ies_bad_phi()` with "absolute bad_phi" and `ies_bad_phi_sigma()` with "adaptive bad_phi"? Then I guess it would solve my specific problem, but I understand the general idea behind how it is implemented now. Probably you could also argue that if the bad_phi realizations have such a huge impact on the data conflicts this is indication to reconsider how the prior is defined (like making sure the pilot point hk distribution around production wells is not yielding to low values). I was mainly curious why it is the way it is, since it did not do what I expected based on what I intended to do.

**Reply by jtwhite79** [reply]
you got it right RE those two phi filters.  I'll add this idea to the next release
